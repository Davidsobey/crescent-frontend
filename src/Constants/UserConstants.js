const UserConstants = {
  REGISTER_REQUEST: 'USERS_REGISTER_REQUEST',
  REGISTER_SUCCESS: 'USERS_REGISTER_SUCCESS',
  REGISTER_FAILURE: 'USERS_REGISTER_FAILURE',

  LOGIN_REQUEST: 'USERS_LOGIN_REQUEST',
  LOGIN_SUCCESS: 'USERS_LOGIN_SUCCESS',
  LOGIN_FAILURE: 'USERS_LOGIN_FAILURE',
  LOGIN_CLOSE: 'USERS_LOGIN_CLOSE',

  LOGOUT: 'USERS_LOGOUT',

  GETALL_REQUEST: 'USERS_GETALL_REQUEST',
  GETALL_SUCCESS: 'USERS_GETALL_SUCCESS',
  GETALL_FAILURE: 'USERS_GETALL_FAILURE',

  GET_ALL_ROLES_REQUEST: 'GET_ALL_ROLES_REQUEST',
  GET_ALL_ROLES_SUCCESS: 'GET_ALL_ROLES_SUCCESS',
  GET_ALL_ROLES_FAILURE: 'GET_ALL_ROLES_FAILURE',

  GETUSER_REQUEST: 'USERS_GETUSER_REQUEST',
  GETUSER_SUCCESS: 'USERS_GETUSER_SUCCESS',
  GETUSER_FAILURE: 'USERS_GETUSER_FAILURE',

  DELETE_REQUEST: 'USERS_DELETE_REQUEST',
  DELETE_SUCCESS: 'USERS_DELETE_SUCCESS',
  DELETE_FAILURE: 'USERS_DELETE_FAILURE',

  EDIT_USER_REQUEST: 'USERS_EDIT_REQUEST',
  EDIT_USER_SUCCESS: 'USERS_EDIT_SUCCESS',
  EDIT_USER_FAILURE: 'USERS_EDIT_FAILURE',

  ENROL_REQUEST: 'ENROL_USER_REQUEST',
  ENROL_SUCCESS: 'ENROL_USER_SUCCESS',
  ENROL_FAILURE: 'ENROL_USER_FAILURE',

  ENROL_ALL_REQUEST: 'ENROL_ALL_REQUEST',
  ENROL_ALL_SUCCESS: 'ENROL_ALL_SUCCESS',
  ENROL_ALL_FAILURE: 'ENROL_ALL_FAILURE',

  EDIT_USER_SUCCESS: 'EDIT_USER_SUCCESS',
};

export default UserConstants;
