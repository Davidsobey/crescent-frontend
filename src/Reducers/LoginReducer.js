import UserConstants from '../Constants/UserConstants';

function UserReducer(state = {}, action) {
  switch (action.type) {
    case UserConstants.LOGIN_REQUEST:
      return {
        loading: true,
        user: action.user,
      };
    case UserConstants.LOGIN_SUCCESS:
      return {
        loggedIn: true,
        user: action.user,
      };
    case UserConstants.LOGIN_FAILURE:
      return {};
    case UserConstants.LOGOUT:
      return {};
    default:
      return state;
  }
}

export default UserReducer;
