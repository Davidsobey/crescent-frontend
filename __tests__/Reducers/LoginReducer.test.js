// Reducer to be tested
import LoginReducer from '../../src/Reducers/LoginReducer';
import UserConstants from '../../src/Constants/UserConstants';

describe('LoginReducer', () => {
  test('LOGIN_SUCCESS', () => {
    const action = { 
      type: UserConstants.LOGIN_SUCCESS, 
      user: {id:1, name:'aaa'},
    };

    expect(LoginReducer(undefined, action)).toMatchSnapshot();
  });
});
