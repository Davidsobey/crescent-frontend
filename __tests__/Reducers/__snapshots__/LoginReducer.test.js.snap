// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginReducer LOGIN_SUCCESS 1`] = `
Object {
  "loading": false,
  "loggedIn": true,
  "user": Object {
    "id": 1,
    "name": "aaa",
  },
}
`;
